crates/turborepo-lockfiles/src/yarn1/de.rs

```diff
@@ -0,0 +1,320 @@
+use std::sync::OnceLock;
+
+use nom::{
+    branch::alt,
+    bytes::complete::{escaped_transform, is_not, tag, take_till},
+    character::complete::{anychar, char as nom_char, crlf, newline, none_of, satisfy, space1},
+    combinator::{map, not, opt, peek, recognize, value},
+    multi::{count, many0, many1},
+    sequence::{delimited, pair, preceded, separated_pair, terminated, tuple},
+    IResult,
+};
+use regex::Regex;
+use serde_json::Value;
+
+// regex for trimming spaces from start and end
+fn pseudostring_replace() -> &'static Regex {
+    static RE: OnceLock<Regex> = OnceLock::new();
+    RE.get_or_init(|| Regex::new(r"^ *| *$").unwrap())
```

Comment:
Not blocking, but maybe consider the `lazy_regex` crate to encapsulate this pattern?